### ISSUE TITLE
BSOD on boot after installing included web server

### ISSUE THREAD

**Issue by bovine3dom** [multiOTP/multiotp#32]
I installed multiotp using the `webservice_install.cmd` and the credential provider on my Windows 10 machine.

I used remote desktop to check that the provider was working (it was).

I then ran `Netplwiz.exe` to ensure that users had to enter a password to log in (I previously had this disabled).

I rebooted the machine to check that the web server would start without the user logging in, and got an "IRQL_NOT_LESS_OR_EQUAL" blue screen of death before I could log in. This happened 3 more times, but with "MEMORY_MANAGEMENT" errors instead.

I booted into safe mode and uninstalled the web server using `webservice_uninstall.cmd`, and restarted the PC. I have not had a blue screen since.

Please find the BSOD minidumps attached: [minidumps.zip](https://github.com/multiOTP/multiotp/files/1834462/minidumps.zip)

I believe two of them are corrupted as they will not open in BlueScreenView.

---

Edit: after reinstalling using `webservice_install.cmd`, everything seems to work fine, no blue screens after reboot. I haven't touched `Netplwiz.exe` though.

**Comment by multiOTP** [comment]
Hello,

Thanks evaluating our multiOTP solution. The windows version of the web server is provided "as is" and is not recommended to be installed for production. You should use a virtual machine installation instead (virtual image is provided).

But the good news is that if you want to protect a single Windows computer, you don't have to install any server on it, because everything concerning strong authentication check is already included in the multiotp.exe file installed and called by the multiOTP Credential Provider.

A multiOTP installed as a server is only needed when several computers and/or devices want to share a common strong two factor authentication server.

Check the readme file and follow the instruction on "**HOW TO INSTALL A LOCAL ONLY STRONG AUTHENTICATION ON A WINDOWS MACHINE ?**"

1) Install multiOTPCredentialProvider, which contains also multiOTP inside.
   It works with Windows 7/8/8.1/10/2012(R2)/2016 in both 32 and 64 bits.
   (http://download.multiotp.net/credential-provider/).
2) During the installation, specify the folder on the client where the
   multiotp.exe file must be installed and configured.
3) In the wizard, leave the URL of the multiOTP server(s) empty.
   multiotp.exe file must be installed and configured.
4) You can also choose to require a strong authentication only for RDP.
5) When you are on the test page, open the folder where multiOTP is installed
   and create a new local user as explained above.
6) If the test is successful, the Credential Provider is installed.
7) To disable the Credential Provider, uninstall it from Windows,
   or execute multiOTPCredentialProvider-unregister.reg

Best regards,

**Comment by bovine3dom** [comment]
Thanks, I can't believe I missed that about not needing a web server. It did seem strange. It works perfectly now, thank you.

I agree that a little web server should not cause a BSOD and that it sounds implausible, but I haven't received a BSOD on this computer ever before. I installed the web server, rebooted within a few minutes, got 4 BSODs, uninstalled the web server from safe mode, and the BSODs stopped happening. That suggests to me that it was the proximate cause, perhaps complicated by the fact that I changed the settings in Netplwiz after installing the server and credential provider but before rebooting.

I would be happy to help you troubleshoot this if you would like, but the problem has gone away for me, so I don't mind as much :)
